odoo/addons/base/models/ir_actions_report.py

```diff
@@ -752,7 +757,7 @@ def _render_qweb_pdf_prepare_streams(self, report_ref, data, res_ids=None):
 
             html = self.with_context(**additional_context)._render_qweb_html(report_ref, all_res_ids_wo_stream, data=data)[0]
 
-            bodies, html_ids, header, footer, specific_paperformat_args = self.with_context(**additional_context)._prepare_html(html, report_model=report_sudo.model)
+            bodies, html_ids, header, footer, specific_paperformat_args = report_sudo.with_context(**additional_context)._prepare_html(html, report_model=report_sudo.model)
```

Comment:
Because the self didn't have any xml_id and so it was impossible to retrieve the paperformat that was used, by getting the report_sudo we can now have the xml_id in the body, header and footer and apply a condition on the css_margin fields